python/python/lance/torch/data.py

```diff
@@ -216,31 +223,16 @@ def __init__(
         self._hf_converter = None
 
         # As Shared Dataset
+        self.shard_granularity = shard_granularity
         self.rank = rank
         self.world_size = world_size
-        self.shard_granularity = shard_granularity
-        if sampler is None:
-            if shard_granularity is None:
-                if rank is not None or world_size is not None:
-                    warnings.warn(
-                        "rank and world_size are deprecated,"
-                        + " use ShardedFragmentSampler instead.",
-                    )
-                    sampler = ShardedFragmentSampler(rank=rank, world_size=world_size)
-                else:
-                    sampler = FullScanSampler()
-            elif shard_granularity == "batch":
-                sampler = ShardedBatchSampler(rank, world_size)
-            elif shard_granularity == "fragment":
-                sampler = ShardedFragmentSampler(rank, world_size)
-            else:
-                raise ValueError("Invalid shard_granularity: {}")
+        if rank is not None and world_size is not None:
+            warnings.warn("rank and world_size are deprecated")
```

Comment:
nit: specify the warning type as https://docs.python.org/3/library/exceptions.html#DeprecationWarning

Reply:
done